pyanaconda/modules/subscription/runtime.py

```diff
@@ -1084,6 +1100,11 @@ def run(self):
 
         :raises: RegistrationError if calling the RHSM DBus API returns an error
         """
+        # check if we can just return cached data
+        if self._cached_org_data_list:
```

Comment:
It seems dangerous to always prefer the cache without trying to get the actual organizations. I wouldn't trust this logic. The cache should be used as only as a fallback when you fail to retrieve the data.

Reply:
Would these really change that often. I'm not sure but this seems to me like a reasonable solution?